zerver/tests/test_auth_backends.py

```diff
@@ -1633,6 +1640,13 @@ def social_auth_test_finish(self, result: HttpResponse,
             # that it's definitely the GitHub authentication backend.
             if self.AUTH_FINISH_URL == "/complete/github/":
                 self.assert_in_success_response(["Select account"], result)
+                # verifies if all the emails returned by GitHub auth
+                # are in "choose email" screen verified emails to
+                # choose and uverified emails for user to see.
+                for email_data_dict in email_data:
+                    email = email_data_dict["email"]
+                    if not email.endswith("noreply.github.com"):
+                        self.assert_in_success_response([], result)
```

Comment:
Should we also assert that if it does end with noreply.github.com it must not appear in the response?

Reply:
It probably is a good thing to do. adding that too.